data_safe_haven/external/api/azure_cli.py

```diff
@@ -68,10 +68,11 @@ def confirm(self) -> None:
             return None
 
         account = self.account
-        self.logger.info(
-            f"name: {account.name} (id: {account.id_}\ntenant: {account.tenant_id})"
-        )
-        if not typer.confirm("Is this the Azure account you expect?\n"):
+        self.logger.info(f"Azure user: {account.name} ({account.id_})")
+        self.logger.info(f"Azure tenant ID: {account.tenant_id})")
+        if not self.logger.confirm(
+            "Is this the Azure account you expect?", default_to_yes=False
+        ):
```

Comment:
Is this still `typer.confirm`?

Reply:
It's a wrapper around `rich.prompt.Confirm`. Is it important to use `typer.confirm`? We run `typer.Exit` if the answer is "no".

Reply:
Not that I'm aware of, you still need to handle the yes/no with `typer.confirm`.

Is there an advantage to using the Rich function?

We should use only one, whichever one we decide.

Reply:
Using the LoggingSingleton functions means that you can redirect the log streams to e.g. files or logging analytics platforms using the standard logging handlers. I think I prefer this over `typer`'s logging commands.